libcodechecker/server/db_cleanup.py

```diff
@@ -0,0 +1,34 @@
+import sqlalchemy
+
+import shared
+from codeCheckerDBAccess_v6 import constants, ttypes
+from codeCheckerDBAccess_v6.ttypes import*
+
+from libcodechecker.logger import LoggerFactory
+from libcodechecker.server.database.run_db_model import *
+
+LOG = LoggerFactory.get_new_logger('DB CLEANUP')
+
+
+def remove_unused_files(session):
+    LOG.info("Garbage collection of dangling files started...")
+
+    bpe_files = session.query(BugPathEvent.file_id).subquery()
```

Comment:
This will generate the following SQL query:
```sql
SELECT file_id FROM bug_path_events
```
If the `bug_path_events` table is large (~1101374 rows), this query will be slow (>60000ms).
You have to `group by` this query by `file_id` or use distinct on file_id:
```sql
SELECT file_id FROM bug_path_events
GROUP BY file_id
```
This will reduce the query time to 390ms and the result to 10042.